library/src/main/java/com/schibsted/spain/barista/interaction/BaristaScrollInteractions.kt

```diff
@@ -16,43 +16,53 @@ import org.hamcrest.Matcher
  */
 object BaristaScrollInteractions {
 
-  // This value has been mathematically calculated and proven to be precisely the exact number of retries needed to always work.
-  // Not really, we just tried hundreds of times with different values and this seems to be the best one.
-  private val MAX_SCROLL_ATTEMPTS = 50
-
-  @JvmStatic
-  fun scrollTo(id: Int) {
-    scrollWithMultipleAttempts(id.resourceMatcher(), failAtEnd = true)
-  }
-
-  @JvmStatic
-  fun scrollTo(text: String) {
-    scrollWithMultipleAttempts(withText(text), failAtEnd = true)
-  }
-
-  @JvmStatic
-  fun safelyScrollTo(id: Int) {
-    scrollWithMultipleAttempts(id.resourceMatcher(), failAtEnd = false)
-  }
-
-  @JvmStatic
-  fun safelyScrollTo(text: String) {
-    scrollWithMultipleAttempts(withText(text), failAtEnd = false)
-  }
-
-  private fun scrollWithMultipleAttempts(matcher: Matcher<View>, failAtEnd: Boolean) {
-    val spyFailureHandler = SpyFailureHandler()
-    for (i in 1..MAX_SCROLL_ATTEMPTS) {
-      try {
-        onView(matcher)
-            .withFailureHandler(spyFailureHandler)
-            .perform(nestedScrollToAction())
-      } catch (exception: PerformException) {
-        if (i == MAX_SCROLL_ATTEMPTS && failAtEnd) {
-          spyFailureHandler
-              .resendLastError("Could not scroll to ${matcher.description()}. Retried ${MAX_SCROLL_ATTEMPTS} times but all failed")
+    // This value has been mathematically calculated and proven to be precisely the exact number of retries needed to always work.
+    // Not really, we just tried hundreds of times with different values and this seems to be the best one.
+    private val MAX_SCROLL_ATTEMPTS = 50
+
+    @JvmStatic
+    fun scrollTo(matcher: Matcher<View>) {
+        scrollWithMultipleAttempts(matcher, failAtEnd = true)
+    }
+
+    @JvmStatic
+    fun scrollTo(id: Int) {
+        scrollWithMultipleAttempts(id.resourceMatcher(), failAtEnd = true)
+    }
+
+    @JvmStatic
+    fun scrollTo(text: String) {
+        scrollWithMultipleAttempts(withText(text), failAtEnd = true)
+    }
+
+    @JvmStatic
+    fun safelyScrollTo(matcher: Matcher<View>) {
```

Comment:
@AdevintaSpain/barista-maintainers we should explain this method in the README or at least write a Javadoc for it...